### PR TITLE
enable new function test framework by default

### PR DESCRIPTION
**Pull request by windtalker**
Signed-off-by: xufei <xufeixw@mail.ustc.edu.cn>

### What problem does this PR solve?

Issue Number: ref #4830

Problem Summary:

### What is changed and how it works?
Currently, function unit test will create function directly but DAGExpressionAnalyzer, it may add some rewrite when compiling function, #4861 add a new function unit test framework, which will convert columns to tipb expression first, and then use `DAGExpressionAnalyzer` to build the function. This pr enable this new function test framework by default.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Functions/tests/gtest_unix_timestamp.cpp [hunk 1]
```diff
@@ -86,17 +86,17 @@ try
...
-        executeFunction(func_name_int, createConstColumn<Nullable<MyDateTime>>(std::make_tuple(0), 10, date_time_data[0])));
+        createConstColumn<UInt64>(10, date_time_int_result[0]),
+        executeFunction(func_name_int, createConstColumn<MyDateTime>(std::make_tuple(0), 10, date_time_data[0])));
     ASSERT_COLUMN_EQ(
-        createConstColumn<Nullable<Decimal64>>(std::make_tuple(12, 0), 10, date_time_decimal_result[0]),
-        executeFunction(func_name_dec, createConstColumn<Nullable<MyDateTime>>(std::make_tuple(0), 10, date_time_data[0])));
+        createConstColumn<Decimal64>(std::make_tuple(12, 0), 10, date_time_decimal_result[0]),
+        executeFunction(func_name_dec, createConstColumn<MyDateTime>(std::make_tuple(0), 10, date_time_data[0])));
     ASSERT_COLUMN_EQ(
-        createConstColumn<Nullable<Decimal64>>(std::make_tuple(18, 6), 10, date_time_with_fsp_result[0]),
```
Comment (SeaRise) [inline dbms/src/Functions/tests/gtest_unix_timestamp.cpp:98]: Can we convert `Const<Nullable<Type>>` to `Const<Type>` in `FunctionTestUtils`, and then we don't need to modify the unit test.

File: dbms/src/Functions/tests/gtest_arithmetic_functions.cpp [hunk 2]
```diff
@@ -128,6 +128,8 @@ try
     null_or_zero_field.push_back(Field(DecimalField<Decimal128>(0, 0)));
 
     std::vector<Int64> values{10, 2, 20, 8, 10, 0, 30, 8, 16, 4};
+    Int128 decimal_128_factor = 10000000000;
+    decimal_128_factor *= 1000000000;
```
Comment (SeaRise) [inline dbms/src/Functions/tests/gtest_arithmetic_functions.cpp:132]: add some comments?

File: dbms/src/Functions/tests/gtest_unix_timestamp.cpp (hunk 1 above)
Comment (windtalker) [reply dbms/src/Functions/tests/gtest_unix_timestamp.cpp:98]: This is an intentional behavior, because in TiDB, it will generate a constant expr with nullable type but not null value, we need to cover this case, if we convert `Const<Nullable<Type>>(not_null_value)` to `Const<Type>(not_null_value)` in `FunctionTestUtils`, we don't have case to cover this.

File: dbms/src/Functions/tests/gtest_arithmetic_functions.cpp (hunk 2 above)
Comment (windtalker) [reply dbms/src/Functions/tests/gtest_arithmetic_functions.cpp:132]: Done

File: dbms/src/Functions/tests/gtest_unix_timestamp.cpp (hunk 1 above)
Comment (SeaRise) [reply dbms/src/Functions/tests/gtest_unix_timestamp.cpp:98]: Did we cover this case in the unit test?
what happens when `createConstColumn<Nullable<Decimal64>>(std::make_tuple(18, 6), 10, date_time_with_fsp_result[0])`?

File: dbms/src/Functions/tests/gtest_bitand.cpp [hunk 3]
```diff
@@ -126,51 +126,40 @@ try
...
 CATCH
 
 TEST_F(TestFunctionBitAnd, Boundary)
 try
 {
-    ASSERT_BITAND(createColumn<Int8>({127, 127, -128, -128}), createColumn<UInt8>({0, 255, 0, 255}), createColumn<Int8>({0, 127, 0, -128}));
-    ASSERT_BITAND(createColumn<Int8>({127, 127, -128, -128}), createColumn<UInt16>({0, 65535, 0, 65535}), createColumn<Int16>({0, 127, 0, -128}));
-    ASSERT_BITAND(createColumn<Int16>({32767, 32767, -32768, -32768}), createColumn<UInt8>({0, 255, 0, 255}), createColumn<Int16>({0, 255, 0, 0}));
+    ASSERT_BITAND(createColumn<Int8>({127, 127, -128, -128}), createColumn<UInt8>({0, 255, 0, 255}), createColumn<UInt64>({0, 127, 0, 128}));
+    ASSERT_BITAND(createColumn<Int8>({127, 127, -128, -128}), createColumn<UInt16>({0, 65535, 0, 65535}), createColumn<UInt64>({0, 127, 0, 65408}));
```
Comment (SeaRise) [inline dbms/src/Functions/tests/gtest_bitand.cpp:157]: Does this meet expectations?
What does `65408` mean?

File: dbms/src/Functions/tests/gtest_bitand.cpp [hunk 4]
```diff
@@ -126,51 +126,40 @@ try
...
+    ASSERT_BITAND(createColumn<Int16>({32767, 32767, -32768, -32768}), createColumn<UInt8>({0, 255, 0, 255}), createColumn<UInt64>({0, 255, 0, 0}));
 
     ASSERT_BITAND(createColumn<Int64>({0, 0, 1, 1, -1, -1, INT64_MAX, INT64_MAX, INT64_MIN, INT64_MIN}),
                   createColumn<UInt64>({0, UINT64_MAX, 0, UINT64_MAX, 0, UINT64_MAX, 0, UINT64_MAX, 0, UINT64_MAX}),
-                  createColumn<Int64>({0, 0, 0, 1, 0, -1, 0, INT64_MAX, 0, INT64_MIN}));
+                  createColumn<UInt64>({0, 0, 0, 1, 0, UINT64_MAX, 0, INT64_MAX, 0, 9223372036854775808ull}));
```
Comment (SeaRise) [inline dbms/src/Functions/tests/gtest_bitand.cpp:162]: ditto for `9223372036854775808ull`

File: dbms/src/Functions/tests/gtest_datetime_daymonthyear.cpp [hunk 5]
```diff
@@ -76,15 +76,7 @@ class TestDateTimeDayMonthYear : public DB::tests::FunctionTest
...
-                column = ColumnWithTypeAndName(
-                    createConstColumn<Nullable<typename DT::FieldType>>(1,
-                                                                        {raw_input.toPackedUInt()})
-                        .column,
-                    makeNullable(std::make_shared<DT>()),
-                    "result");
-                result_day = createConstColumn<Nullable<UInt8>>(1, {day});
-                result_month = createConstColumn<Nullable<UInt8>>(1, {month});
-                result_year = createConstColumn<Nullable<UInt16>>(1, {year});
+                return;
```
Comment (SeaRise) [inline dbms/src/Functions/tests/gtest_datetime_daymonthyear.cpp:79]: or throw exception?

Comment (windtalker) [reply dbms/src/Functions/tests/gtest_datetime_daymonthyear.cpp]: I add these tests back.

File: dbms/src/Functions/tests/gtest_unix_timestamp.cpp (hunk 1 above)
Comment (windtalker) [reply dbms/src/Functions/tests/gtest_unix_timestamp.cpp:98]: Oh, my mistake, at the beginning I just remove all the tests withe `Const<Nullable<Type>>(not_null_value)`, then I realized that I need to keep these tests, because TiDB will geneate a constant expr with nullable type but not null value. So I decide to updated these tests' result instead of just remove them. Unfortunately I forgot to update the tests here.

**Review by SeaRise — APPROVED** [review]
